src/coverlet.msbuild.tasks/CoverageResultTask.cs

```diff
@@ -76,16 +77,18 @@ public string CoverletMultiTargetFrameworksCurrentTFM
 
         public CoverageResultTask()
         {
-            _logger = Services.Current.GetService<ILogger>();
+           _serviceProvider = new Services().GetServiceProvider();
```

Comment:
MsbuildLogger needs TaskLogginHelper so you need to pass it to `GetServiceProvider`  https://github.com/tonerdo/coverlet/blob/63dbda8190c505d293c7266445143b11b02f56e9/src/coverlet.msbuild.tasks/MSBuildLogger.cs#L15

Reply:
The same for InstrumentationTask

Reply:
Done